Scheduler: describe selectionEnd APIs (#8695)
DevExpress/devextreme-documentation: api-reference/10 UI Components/dxScheduler/1 Configuration/onSelectionEnd.md

---
---
##### shortDescription
A function that is executed when a user finishes selecting a date-time range.

##### param(e): ui/scheduler:SelectionEndEvent
Information about the event.

##### field(e.component): {WidgetName}
The UI component's instance.

##### field(e.element): DxElement
#include common-ref-elementparam with { element: "UI component" }

##### field(e.selectedCellData): Array<any>

> An array of objects describing the selected cells. Each object contains `startDate`, `endDate`, `allDay`, and `groups` properties.

Reviewer: ```suggestion
An object array that identifies selected cells. Each array element contains the following properties: `startDate`, `endDate`, `allDay`, and `groups`.
```

---

> This event fires after the user releases the pointer button following a drag across one or more scheduler cells. Use it to respond to a completed cell range selection (for example, to open the appointment popup pre-filled with the selected time range).

Reviewer: I think we can refer to a "drag-and-drop selection operation" and that should be clear and would allow not to go into details like "releases the button following a drag across". 

```suggestion
This event fires after the user finishes a drag-and-drop selection operation. Handle the event to respond to a completed cell range selection (for example, to open the appointment popup pre-filled with the selected time range).
```

> The handler receives a [selectedCellData](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#selectedCellData) array. Each item identifies a selected cell using the following properties: `startDate`, `endDate`, `allDay`, and `groups`. To create an appointment that spans the selected range, pass the first cell's `startDate` and the last cell's `endDate` to [showAppointmentPopup](/Documentation/ApiReference/UI_Components/dxScheduler/Methods/#showAppointmentPopupappointmentData_createNewAppointment_currentAppointmentData).

Reviewer 2: ```suggestion
The handler receives a [selectedCellData](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#selectedCellData)  array. Each item identifies a selected cell with the following properties: `startDate`, `endDate`, `allDay`, and `groups`. To create an appointment for the selected range, create an `appointmentData` object with the first cell’s `startDate` and the last cell’s `endDate`, then pass this object to [showAppointmentPopup](/Documentation/ApiReference/UI_Components/dxScheduler/Methods/#showAppointmentPopupappointmentData_createNewAppointment_currentAppointmentData).
```

> The handler receives a [selectedCellData](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#selectedCellData) array whose items contain `startDate`, `endDate`, `allDay`, and `groups` properties describing each cell in the selection. To create an appointment that spans the selected range, pass the first cell's `startDate` and the last cell's `endDate` to [showAppointmentPopup](/Documentation/ApiReference/UI_Components/dxScheduler/Methods/#showAppointmentPopupappointmentData_createNewAppointment_currentAppointmentData).

Reviewer: ```suggestion
The handler receives a [selectedCellData](/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#selectedCellData) array. Each item identifies a selected cell using the following properties: `startDate`, `endDate`, `allDay`, and `groups`. To create an appointment that spans the selected range, pass the first cell's `startDate` and the last cell's `endDate` to [showAppointmentPopup](/Documentation/ApiReference/UI_Components/dxScheduler/Methods/#showAppointmentPopupappointmentData_createNewAppointment_currentAppointmentData).
```

[note]

> - The event does not fire when the user clicks on a cell that is already selected.

Reviewer 2: ```suggestion
- The event does not fire when users click on a cell that is already selected.
```
